fam/llm/fast_inference_utils.py

```diff
@@ -290,11 +278,33 @@ def _load_model(checkpoint_path, spk_emb_ckpt_path, device, precision):
             k = k.replace(".mlp.c_proj.", ".feed_forward.w2.")
 
     model.load_state_dict(state_dict, assign=True)
-    # simple_quantizer = WeightOnlyInt8QuantHandler(model)
-    # quantized_state_dict = simple_quantizer.create_quantized_state_dict()
-    # model = simple_quantizer.convert_for_runtime()
-    # model.load_state_dict(quantized_state_dict, assign=True)
-    model = model.to(device=device, dtype=precision)
+    model = model.to(device=device, dtype=torch.bfloat16)
+
+    if quantisation_mode == "int8":
+        warnings.warn(
+            "int8 quantisation is slower than bf16/fp16 for undebugged reasons! Please set optimisation_mode to `None` or to `int4`."
+        )
+        warnings.warn(
+            "quantisation will degrade the quality of the audio! Please set optimisation_mode to `None` for best quality."
+        )
+        simple_quantizer = WeightOnlyInt8QuantHandler(model)
+        quantized_state_dict = simple_quantizer.create_quantized_state_dict()
+        model = simple_quantizer.convert_for_runtime()
+        model.load_state_dict(quantized_state_dict, assign=True)
+        model = model.to(device=device, dtype=torch.bfloat16)
+        torch.cuda.empty_cache()
```

Comment:
Did this end up solving the issue with retained memory pre quantization?

Reply:
Nope! Will add a TODO